syntax = "proto3";

message Piece {
    bytes symbolicData = 1;
    string encoding = 2;
    string name = 3;
}

message Note {
    float onset = 1;
    float offset = 2;
    int32 pitchB40 = 3;

    uint32 pieceIdx = 6;

    uint32 measureNumber = 5; 
    uint32 measureIdx = 7;
}

message Measure {
    bytes symbolicData = 1;
    uint32 number = 2;
}

message Vector {
    Note left = 1;
    Note right = 2;
}